### PR TITLE
fix: fix dragHandle issue

### PR DESCRIPTION
**Pull request by burakukula**
Temporary fix to the issue that we face in the field editors.
The library used by field editors (react-sortable-hoc) is canceling events for buttons for some reason (https://github.com/clauderic/react-sortable-hoc/blob/d94ba3cc67cfc7d6d460b585e7723bdb50015e53/src/SortableContainer/defaultShouldCancelStart.js).
We should change the implementation on the field editors at some point. We will have a look at it while working on the keyboard nav for dragHandle (inspiration from user interface, where it's already done)

I added the example in the storybook,but skipped the website, we don't want to encourage anyone to use react-sortable-hoc, but maybe it's still worth keeping the example in the storybook?

### CODE REVIEW

File: packages/components/drag-handle/src/DragHandle.tsx [hunk 1]
```diff
@@ -107,7 +107,9 @@ export const DragHandle = forwardRef<
     );
 
     return (
-      <button
+      <div
```
Comment (suevalov) [inline packages/components/drag-handle/src/DragHandle.tsx:110]: 🤯

**Review by suevalov — APPROVED** [review]

File: packages/components/drag-handle/src/DragHandle.tsx (hunk 1 above)
Comment (burakukula) [reply packages/components/drag-handle/src/DragHandle.tsx:110]: 🤷‍♀️

File: packages/components/drag-handle/src/DragHandle.tsx [hunk 2]
```diff
@@ -6,7 +6,7 @@ import React, {
   MouseEventHandler,
 } from 'react';
 import { cx } from 'emotion';
-import type { PropsWithHTMLElement } from '@contentful/f36-core';
+import type { PropsWithHTMLDivElement } from '@contentful/f36-core';
```
Comment (bgutsol) [inline packages/components/drag-handle/src/DragHandle.tsx:9]:
```suggestion
import type { PropsWithHTMLElement } from '@contentful/f36-core';
```

File: packages/components/drag-handle/src/DragHandle.tsx [hunk 3]
```diff
@@ -32,13 +32,13 @@ export type DragHandleInternalProps = CommonProps & {
   label: string;
 };
 
-export type DragHandleProps = PropsWithHTMLElement<
+export type DragHandleProps = PropsWithHTMLDivElement<
```
Comment (bgutsol) [inline packages/components/drag-handle/src/DragHandle.tsx:35]:
```suggestion
export type DragHandleProps = PropsWithHTMLElement<
```

**Review by bgutsol — COMMENTED** [review]
Small typo

File: packages/components/drag-handle/src/DragHandle.tsx (hunk 2 above)
Comment (bgutsol) [reply packages/components/drag-handle/src/DragHandle.tsx]: I'm curious why u didn't get an TS error here

Comment (burakukula) [reply packages/components/drag-handle/src/DragHandle.tsx]: fixed, yeah, dunno why it didn't catch it. Maybe I need to restart. Anyway, fixed. Thx!

**Review by bgutsol — APPROVED** [review]

**Pull request merged**